client/signup/main.jsx

```diff
@@ -207,6 +208,11 @@ class Signup extends Component {
 				)
 			);
 		}
+
+		// preload the experiment to minimize the perceived loading time
+		if ( this.props.flowName === flows.defaultFlowName ) {
+			loadExperimentAssignment( 'calypso_plans_2yr_toggle' );
```

Comment:
It is necessary to have this conditional on `is2023PricingGridVisible`.

Reply:
> I originally think it's find to preload here since there are exposing events for filtering out unwanted traffic. If it's not safe, maybe I should just not preload it?

This isn't ideal, but it is ok.

Reply:
> I originally think it's find to preload here since there are exposing events for filtering out unwanted traffic. If it's not safe, maybe I should just not preload it?

I'd say it's safe to preload like this. Then, using the exposure event with `path: /start/plans` will filter out all users that were assigned and dropped before reaching that step.